src/Containers/Microsoft.NET.Build.Containers/Registry/Registry.cs

```diff
@@ -27,22 +27,26 @@ internal sealed class Registry
     /// This is used in user-facing error messages, and it should match what the user would manually enter as
     /// part of Docker commands like `docker login`.
     /// </summary>
-    public string RegistryName { get; init; }
+    public string RegistryName { get; }
 
-    internal Registry(Uri baseUri, ILogger logger) : this(baseUri, logger, new DefaultRegistryAPI(baseUri, logger), new RegistrySettings()) { }
+    internal Registry(string registryName, ILogger logger, IRegistryAPI? registryAPI = null, RegistrySettings? settings = null) :
```

Comment:
This new constructor accepts the `string registryName` that we get from the user.